constants/src/main/java/org/jboss/pnc/constants/Patterns.java

```diff
@@ -29,7 +29,7 @@
      * For example 1.0.0.ER1, 1.2.10.CR1, 1.0.0.CD1.CR1
      * See {@link org.jboss.pnc.constants.PatternsTest} for valid examples.
      */
-    public static final String PRODUCT_MILESTONE_VERSION = "^[0-9]+\\.[0-9]+\\.[0-9]+\\.[\\w\\.-]+$";
+    public static final String PRODUCT_MILESTONE_VERSION = "^[0-9]+\\.[0-9]+(\\.[0-9]+)?\\.([a-zA-Z][\\w-]*)+(\\.[a-zA-Z][\\w-]*)?$";
```

Comment:
Can't it be like this?
```suggestion
    public static final String PRODUCT_MILESTONE_VERSION = "^[0-9]+\\.[0-9]+(\\.[0-9]+)?\\.([a-zA-Z][\\w.-]*)+?$";
```

Reply:
this would make us accept more then 2 qualifiers, after discussions woth @jbartece we agreed not to support more than 2